storage/innobase/handler/ha_innodb.cc

```diff
@@ -21437,12 +21448,36 @@ innodb_temp_tablespace_encryption_update(
 @param[in]	save	immediate result from check function */
 static
 void
-innodb_redo_encryption_update(
+update_innodb_redo_log_encrypt(
 	THD*				thd,
 	struct st_mysql_sys_var*	var,
 	void*				var_ptr,
 	const void*			save)
 {
+	const ulong target = *static_cast<const ulong *>(save);
+
+	if (srv_redo_log_encrypt == target) {
+		/* No change */
+		return;
+	}
+
+	if (target == REDO_LOG_ENCRYPT_OFF) {
+  		srv_redo_log_encrypt = REDO_LOG_ENCRYPT_OFF;
+ 		return;
+	}
+
+	if (srv_redo_log_encrypt != REDO_LOG_ENCRYPT_OFF
```

Comment:
Yes, but this is only a partial fix, which I actually encountered because there was a related typo in one of the existing tests (which I rerecorded in this pull request).

There's also the case when the setting is set to a different encryption mode than what's in the currently existing redo log, which I didn't include in this commit (that's PS-5149). I'll add a testcase along with that fix. (I can move this check to that diff too, if that makes more sense).

Reply:
Ok to be deferred to PS-5149